hpke/util.go

```diff
@@ -24,6 +24,23 @@ func (st state) keySchedule(ss, info, psk, pskID []byte) (*encdecContext, error)
 
 	secret := st.labeledExtract(ss, []byte("secret"), psk)
 
+	exporterSecret := st.labeledExpand(
+		secret,
+		[]byte("exp"),
+		keySchCtx,
+		uint16(st.kdfID.Hash().Size()),
+	)
+
+	// XXX This is a hack that is only meant to get the test vectors to pass.
+	// Because the AEAD scheme isn't set, calling Seal() or Open() will panic. I
+	// What I think we want is an alternative code path that returns a
+	// *expContext only.
+	if st.Suite.isExportOnly() {
```

Comment:
What about returining an Exporter interface much earlier.
Currently the flow to get an Opener/Sealer is:
`id -> newSuite() -> NewSender() -> Setup() -> Sealer` 
can we do something simpler for Exporter
`id -> NewSuite() -> NewExporter() -> Exporter `

Reply:
That's a great idea!